templates/lxc-local.in

```diff
@@ -169,8 +169,30 @@ create_build_dir() {
   echo "Using local temporary directory ${LOCAL_TEMP}"
 }
 
-# Unpack the rootfs
-echo "Unpacking the rootfs"
+process_excludes() {
+  # If the file ${LXC_PATH}/excludes exists, it will be used as a list of
+  # files to exclude from the tarball.
+  excludelist=$(relevant_file excludes)
+  if [ -f "${excludelist}" ]; then
+    while read -r line; do
+      EXCLUDES="${EXCLUDES} --exclude=${line}"
+    done < "${excludelist}"
+  fi
+}
+
+unpack_rootfs() {
+  # Unpack the rootfs
+  echo "Unpacking the rootfs to: ${LXC_ROOTFS}"
+
+  # Do not surround ${EXCLUDES} by quotes. This does not work. The solution could
+  # use array but this is not POSIX compliant. The only POSIX compliant solution
+  # is to use a function wrapper, but the latter can't be used here as the args
+  # are dynamic. We thus need to ignore the warning brought by shellcheck.
+  # shellcheck disable=SC2086
+  tar --anchored ${EXCLUDES} --numeric-owner -xpJf "${LXC_FSTREE}" -C "${LXC_ROOTFS}"
+
+  prepare_rootfs
```

Comment:
What's this? That function isn't defined at this stage and also isn't part of the code being moved to the new unpack_rootfs function.

Reply:
I'll try to order the commits more properly next time. Doing this, I worked using my final product and tried to re-do the changes I made so they were associated with individual commits. In the future I will likely just commit one change at a time, in order.